### DISCUSSION TITLE
Optimal memory when converting large files

### DISCUSSION THREAD

**Discussion by mobotpsmocko** [dlemstra/Magick.NET#1599] [Help]
First I want to thank you on such a great product. My question is about optimizing memory for converting large tif files to jp2. I did find that if I set the MemoryResourceLimit to 6gb that it worked much better. I also had our sys admins to add more memory to the server we are using and now everything seems happy but now only using half the memory on the box (which is not bad). Does this setting only set for that instance of Magick.net or is it more of a global setting? Secondly, what would be the optimal setting for this value to get best results. I am using Magick.Native-Q16-x64. Thanks in advance.

**Comment by dlemstra** [comment]
Can you clarify what you mean by a "that instance of Magick.net"? And I also do understand your other question, I probably need a bit more information?

**Reply by mobotpsmocko** [reply]
My windows service queues multiple instances of a image converter service that uses Magick.net to convert images to jp2. So my question is... does setting the memory limit only work for a single instance in use by the instance of of my service or does it effect all instances of Magick.net that would be running.

**Reply by dlemstra** [reply]
It will only affect a single service. But keep in mind that when you run your service in something like IIS with a shared application pool the limit will affect all services.
